pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -528,6 +531,50 @@ func publishMetrics(ctx *zedagentContext, iteration int) {
 	}
 	log.Tracef("DirPaths in persist, elapse sec %v", time.Since(startPubTime).Seconds())
 
+	// ZFS metrics
+	if vault.ReadPersistType() == types.PersistZFS {
+		zpoolList, err := libzfs.PoolOpenAll()
+		if err != nil {
+			log.Errorf("get zpool list failed %v", err)
+		} else {
+			for _, zpool := range zpoolList {
+				rStorageMetric := new(zmet.StorageMetric)
+				rStorageMetric.PoolName = zpool.Properties[libzfs.PoolPropName].Value
+				rStorageMetric.StorageState = zfs.GetZfsDeviceStatusFromStr(zpool.Properties[libzfs.PoolPropHealth].Value)
+				rStorageMetric.CountVolume, err = zfs.GetZfsCountVolume(zpool.Properties[libzfs.PoolPropName].Value)
+				if err != nil {
+					log.Errorf("get count volume failed %v", err)
+				}
+
+				vdevs, err := zpool.VDevTree()
+				if err != nil {
+					log.Errorf("get info about disks in zpool failed %v", err)
+				} else {
+					for _, vdev := range vdevs.Devices {
+						// If this is a RAID or mirror, look at the disks it consists of
+						if vdev.Type == libzfs.VDevTypeMirror || vdev.Type == libzfs.VDevTypeRaidz {
+							for _, disk := range vdev.Devices {
+								rDiskStatus, err := zfs.GetZfsDiskAndStatus(disk)
+								if err == nil {
+									rStorageMetric.Disks = append(rStorageMetric.Disks, rDiskStatus)
+								}
+							}
+							break // for one pool, have one RAID
```

Comment:
That is the case now, but can we make this code handle a potential future case with multiple concatenated RAIDs in a pool?

Reply:
ok